testing/test_monkeypatch.py

```diff
@@ -192,6 +194,40 @@ def test_delenv():
             del os.environ[name]
 
 
+@pytest.mark.skipif(six.PY3, reason="python 2 only bug")
+class TestPython2BytesEnvironKeys(object):
+    """
+    In Python 2, os.environ keys need to be bytes, otherwise this will cause problems with other modules (notably
+    subprocess).
+    """
+
+    VAR_NAME = "PYTEST_INTERNAL_MY_VAR"
+
+    @pytest.fixture(autouse=True)
+    def cleanup(self):
+        self.check_all_environ_keys_are_bytes()
+        yield
+        if self.VAR_NAME in os.environ:
+            del os.environ[self.VAR_NAME]
+
+    def check_all_environ_keys_are_bytes(self):
+        for k in os.environ:
+            assert type(k) is bytes
+
+    def test_setenv_unicode_key(self):
+        monkeypatch = MonkeyPatch()
+        monkeypatch.setenv(unicode(self.VAR_NAME), "2")  # noqa
```

Comment:
instead of calling `unicode`, set the original variable as `b"PYTEST_INTERNAL_MY_VAR"` and use `.decode()` here

Reply:
No problem, but wouldn't that be the same thing in the end? (this test only runs on Python 2 anyway)

Reply:
ah yeah but then:

1. it's explicit the test _intends_ to use bytes objects
2. you don't need `noqa`

Reply:
why not just `u"..."` to begin with - python3.4+ supports that

Reply:
oh yeah, @RonnyPfannschmidt's suggestion is better IMHO